File: Python/Peak Element.py
from typing import List
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        left=0
        right=len(nums)-1
        while left<right:
            mid=(left+right)//2
            if nums[mid]>nums[mid+1]:
                right=mid
            else:
                left=mid+1
        return left


s=Solution()
nums=list(map(int,input("nums : ").split()))
res=s.findPeakElement(nums)
print("Index of peak element : ",res)
print("Peak Element value : ",nums[res])

File: LeetCode/Java/PeakElement.java
import java.util.Scanner;

public class PeakElement {
    public int findPeakElement(int[] nums) {
        int left=0;
        int right=nums.length-1;
        while(left<right){
            int mid=(left+right)/2;
            if(nums[mid]>nums[mid+1]){
                right=mid;
            }
            else{
                left=mid+1;
            }
        }
        return left;
    }
    public static void main(String[] args) {
        Scanner sc =new Scanner(System.in);
        System.out.print("nums : ");
        String[] input=sc.nextLine().split(" ");
        int[] array = new int[input.length];
            for (int j = 0; j < input.length; j++) {
                array[j] = Integer.parseInt(input[j]);
            }
            PeakElement obj = new PeakElement();
        int peakIndex = obj.findPeakElement(array);
        System.out.println("Peak element index: " + peakIndex);
        System.out.println("Peak element value: " + array[peakIndex]);
    }
}